src/NuGet.Client.BaseTypes/SourceRepository.cs

```diff
@@ -71,7 +71,7 @@ public virtual PackageSource PackageSource
         /// </summary>
         /// <typeparam name="T">Expected resource type</typeparam>
         /// <returns>Null if the resource does not exist</returns>
-        public virtual T GetResource<T>()
+        public virtual async Task<T> GetResource<T>()
```

Comment:
GetResourceAsync already exists. There cannot be two Async methods for the same thing.